mart/attack/adversary.py

```diff
@@ -164,12 +163,12 @@ def configure_gradient_clipping(
                 self.gradient_modifier(group["params"])
 
     @silent()
-    def forward(self, *, batch: torch.Tensor | list | dict, model: Callable):
-        # TODO: We may see different batch format when attacking models outside MART. Add a batch_transform?
-        input, target = batch
-
-        # The attack needs access to the model at every iteration.
-        batch_and_model = (input, target, model)
+    def fit(self, *, batch: torch.Tensor | list | dict, model: Callable):
+        # Extract and canonicalize input for initializing perturbation.
+        # TODO: Get rid of batch_c15n() here by converting perturbation to UninitializedParameter.
```

Comment:
This may not be trivial for non-tensor input, such as list and dictionary.

Reply:
Yeah you can't get rid of it here I think.

Reply:
Well maybe you can use `on_train_batch_start` when `batch_idx == 0` to call `self.configure_perturbation()`?